Assets/Huawei/Demos/ML/LanguageDetection/LanguageDetectionDemoManager.cs

```diff
@@ -0,0 +1,188 @@
+using System.Linq;
+using UnityEngine;
+using UnityEngine.UI;
+
+public class LanguageDetectionDemoManager : MonoBehaviour
+{
+
+    private const string Tag = "[HMS] LanguageDetectDemoManager ";
+    public Button onlineButton;
+    public Button offlineButton;
+    public InputField langInputField;
+    public GameObject detectionMenu;
+    public GameObject buttonMenu;
+    public Button firstBestDetectButton;
+    public Button probilityDetectButton;
+    public Button backButton;
+    public Text resultText;
+    public Text titleText;
+    // Start is called before the first frame update
+    void Start()
+    {
+        Debug.Log("AZIZ WAS HERE!!");
+        onlineButton.onClick.AddListener(OnOnlineButtonClicked);
+        offlineButton.onClick.AddListener(OnOfflineButtonClicked);
+        backButton.onClick.AddListener(OnBackButtonClicked);
+        langInputField.text = "Merhaba nasilsin, Hello how are you, مرحبا كيف حالك, 你好吗, Hallo hoe is het, Hola, cómo estás";
+    }
+
+
+    private void OnOnlineButtonClicked()
+    {
+        Debug.Log("Online clicked");
+        OnDetectionButtonClicked(DetectionMode.Online);
+        // Online butonuna basıldığında yapılacak işlemleri ekleyin
+    }
+
+    private void OnOfflineButtonClicked()
+    {
+        Debug.Log("Offline clicked");
+        OnDetectionButtonClicked(DetectionMode.Offline);
+        // Offline butonuna basıldığında yapılacak işlemleri ekleyin
+    }
+    private void OnBackButtonClicked()
+    {
+        Debug.Log("Back clicked");
+        buttonMenu.gameObject.SetActive(true);
+        detectionMenu.gameObject.SetActive(false);
+        resultText.text = string.Empty;
+        titleText.text = string.Empty;
+        firstBestDetectButton.onClick.RemoveAllListeners();
+        probilityDetectButton.onClick.RemoveAllListeners();
+        StopDetection();
+    }
+
+
+    private void OnDetectionButtonClicked(DetectionMode detectionMode)
+    {
+        buttonMenu.gameObject.SetActive(false);
+        detectionMenu.gameObject.SetActive(true);
+        if (detectionMode == DetectionMode.Online)
+        {
+            Debug.Log("Detection ONLINE");
+            titleText.text = "ONLINE DETECTION";
+            HMSMLLanguageDetectionManager.Instance.Configure(detectionMode);
+            firstBestDetectButton.onClick.AddListener(SyncFirstBestDetectOnline);
+            probilityDetectButton.onClick.AddListener(SyncProbilityDetectOnline);
+
+        }
+        else
+        {
+            Debug.Log("Detection OFFLINE");
+            titleText.text = "LOCAL DETECTION";
+            HMSMLLanguageDetectionManager.Instance.Configure(detectionMode);
+            firstBestDetectButton.onClick.AddListener(SyncFirstBestDetectOffline);
+            probilityDetectButton.onClick.AddListener(SyncProbilityDetectOffline);
+        }
+    }
+
+
+    //Ml Remote Lang Detector Functions (Online) 
+    private void FirstBestDetectOnline()
+    {
+        Debug.Log("FirstBestDetectOnline");
```

Comment:
Don't need to double log. Only one enough.

`        Debug.Log("FirstBestDetectOnline" + langInputField.text);
`